src/components/Portal/index.tsx

```diff
@@ -0,0 +1,7 @@
+import { ReactNode } from 'react';
+import ReactDOM from 'react-dom';
+
+export default function Portal({ children }: { children: ReactNode }) {
+  const el = document.getElementById('root');
+  return el && ReactDOM.createPortal(children, el);
```

Comment:
Can you creat another root element to render Portal for modal element?

And I think you might assert type of root element to `HTMLElement`.